mpt/impl/smtp/cassandra-rabbitmq-object-storage/src/test/java/org/apache/james/mpt/smtp/CassandraRabbitMQAwsS3SmtpTestRuleFactory.java

```diff
@@ -40,13 +46,35 @@
 import org.apache.james.util.Host;
 import org.junit.rules.TemporaryFolder;
 
+import com.google.inject.AbstractModule;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.name.Names;
+
 public final class CassandraRabbitMQAwsS3SmtpTestRuleFactory {
     public static SmtpTestRule create(SmtpServerConnectedType smtpServerConnectedType, Host cassandraHost, DockerAwsS3TestRule awsS3TestRule) {
         SmtpTestRule.ServerBuilder createJamesServer = (folder, dnsService) -> createJamesServer(cassandraHost, awsS3TestRule, folder, dnsService);
 
         return new SmtpTestRule(smtpServerConnectedType, createJamesServer);
     }
 
+    private static Module BLOB_STORE_MODULE = new AbstractModule() {
+        @Override
+        protected void configure() {
+            install(new ObjectStorageDependenciesModule());
+            bind(BlobStore.class)
+                .annotatedWith(Names.named(CachedBlobStore.BACKEND))
+                .to(ObjectStorageBlobStore.class);
+        }
+
+        @Provides
+        @Named(MetricableBlobStore.BLOB_STORE_IMPLEMENTATION)
+        @Singleton
+        BlobStore provideBlobStore(@Named(CachedBlobStore.BACKEND) BlobStore blobStore) {
```

Comment:
I think you can bind the BlobStore in `configure` to two annotations

Reply:
I found

```

            bind(BlobStore.class)
                .annotatedWith(Names.named(MetricableBlobStore.BLOB_STORE_IMPLEMENTATION))
                .to(Key.get(BlobStore.class, Names.named(CachedBlobStore.BACKEND)));
```